servicetalk-http-api/src/main/java/io/servicetalk/http/api/ProxyConfig.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2023 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import io.servicetalk.transport.api.ClientSslConfig;
+
+import java.util.function.Consumer;
+
+/**
+ * Configuration for a proxy.
+ *
+ * @param <A> the type of address
+ */
+public interface ProxyConfig<A> {
```

Comment:
For others who review: this may look like an overkill to add a new "config" API while we need only one more config parameter, which could be a 2nd argument for original `proxyAddress(...)` method, like it was implemented in #2698. However, we discussed with @daschl and agreed that having a config object now (similar to `RedirectConfig`) will minimize API changes on `SingleAddressHttpClientBuilder ` in the future, when we may need to add more options, like:
1. Timeout for `CONNECT` request, similar to what Netty does: https://github.com/netty/netty/blob/7e18e8344819428a5560064ac15a2f035b4728fd/handler-proxy/src/main/java/io/netty/handler/proxy/ProxyHandler.java#L57
2. SOCKS proxy config options

Note that `SingleAddressHttpClientBuilder` is used by users when they rely on `HttpProviders`. So, it's important to minimize API friction for them.